src/handlers/transformation/handler.ts

```diff
@@ -0,0 +1,45 @@
+import { S3Event } from "aws-lambda";
+import { readJsonFromS3 } from "../../shared/utils";
+import { configStackName } from "../int-test-support/helpers/envHelper";
+
+import { transformCsvToJson } from "./transform-csv-to-json";
+import { transformRow } from "./transform-row";
+
+export const handler = async (event: S3Event): Promise<void> => {
+  console.log("**event=", event);
+  const outputQueueUrl = process.env.OUTPUT_QUEUE_URL;
+
+  if (outputQueueUrl === undefined || outputQueueUrl.length === 0) {
+    throw new Error("Output queue URL not set.");
+  }
+  console.log("**outputQueueUrl=", outputQueueUrl);
+
+  try {
+    // Set up dependencies
+
+    const idpClientLookup = await readJsonFromS3(
+      configStackName(),
+      "idp_clients/idp-clients.json"
+    );
+    console.log("**idpClientLookup=", idpClientLookup);
+
+    const eventNameRules = await readJsonFromS3(
+      configStackName(),
+      "idp_event_name_rules/idp-event-name-rules.json"
+    );
+    console.log("**eventNameRules=", eventNameRules);
+    const rows = await transformCsvToJson(event);
+    console.log("**rows=", rows);
+
+    // Transform data and send to SQS
+
+    const promises = rows.map(async (row) => {
+      await transformRow(row, idpClientLookup, eventNameRules, outputQueueUrl);
+    });
+
+    await Promise.all(promises);
```

Comment:
Consider `Promise.allSettled()` as `Promise.all()` aborts on the first failure it detects, whereas `allSettled()` waits for all promises to complete before returning.

Also since this could be a very large amount in Promises, processing them in batches, or if the output order matters, sequentially might be an option.

Reply:
Used Promise.allSettled